src/wallet/coincontrol.cpp

```diff
@@ -39,7 +39,9 @@ std::optional<CTxOut> CCoinControl::GetExternalOutput(const COutPoint& outpoint)
 
 PreselectedInput& CCoinControl::Select(const COutPoint& output)
 {
-    return m_selected[output];
+    auto& input = m_selected[output];
+    input.SetPosition(m_selection_pos++);
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/25273/commits/e53789afe60ddbcc014931467ae6b4cab15d0c67:

TIL `m_var++` passes the value, then increments.

nit: I think it's considered more idiomatic (and more readable imo) to do something like:

```suggestion
    input.SetPosition(m_selection_pos);
    m_selection_pos++;
```

Reply:
Done